pkg/cloud/gcp/actuators/util/gcp_uefi_disk_check_test.go

```diff
@@ -0,0 +1,212 @@
+package util_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	machinev1 "github.com/openshift/api/machine/v1beta1"
+	computeservice "github.com/openshift/machine-api-provider-gcp/pkg/cloud/gcp/actuators/services/compute"
+	"github.com/openshift/machine-api-provider-gcp/pkg/cloud/gcp/actuators/util"
+)
+
+var _ = Describe("IsUEFICompatible", func() {
+	var (
+		computeService computeservice.GCPComputeService
+		providerSpec   *machinev1.GCPMachineProviderSpec
+
+		compatible bool
+		err        error
+	)
+
+	BeforeEach(func() {
+		_, computeService = computeservice.NewComputeServiceMock()
+		providerSpec = &machinev1.GCPMachineProviderSpec{}
+	})
+
+	type standardImageInput struct {
+		boot                 bool
+		image                string
+		projectID            string
+		zone                 string
+		expectedErrSubstring string
+		compatible           bool
+	}
```

Comment:
If you added a `modifyProviderSpec func(*machinev1.GCPProviderSpec)`, option here, you could get all the tests in the one pattern, or, you could even just provider the disks directly `disks []*machinev1.GCPDisk` and construct that in each entry?